opentelemetry-sdk/src/opentelemetry/sdk/_metrics/__init__.py

```diff
@@ -51,48 +55,94 @@ class Meter(APIMeter):
     def __init__(
         self,
         instrumentation_info: InstrumentationInfo,
-        meter_provider: APIMeterProvider,
+        measurement_consumer: MeasurementConsumer,
     ):
         super().__init__(instrumentation_info)
         self._instrumentation_info = instrumentation_info
-        self._meter_provider = meter_provider
+        self._measurement_consumer = measurement_consumer
 
     def create_counter(self, name, unit=None, description=None) -> APICounter:
-        return Counter(self._instrumentation_info, name, unit, description)
+        return Counter(
+            self._instrumentation_info,
+            name,
+            self._measurement_consumer,
+            unit,
+            description,
+        )
 
     def create_up_down_counter(
         self, name, unit=None, description=None
     ) -> APIUpDownCounter:
         return UpDownCounter(
-            self._instrumentation_info, name, unit, description
+            self._instrumentation_info,
+            name,
+            self._measurement_consumer,
+            unit,
+            description,
         )
 
     def create_observable_counter(
         self, name, callback, unit=None, description=None
     ) -> APIObservableCounter:
-        return ObservableCounter(
-            self._instrumentation_info, name, callback, unit, description
+
+        instrument = ObservableCounter(
+            self._instrumentation_info,
+            name,
+            self._measurement_consumer,
+            callback,
+            unit,
+            description,
         )
 
+        self._measurement_consumer.register_asynchronous_instrument(instrument)
+
+        return instrument
+
     def create_histogram(
         self, name, unit=None, description=None
     ) -> APIHistogram:
-        return Histogram(self._instrumentation_info, name, unit, description)
+        return Histogram(
+            self._instrumentation_info,
+            name,
+            self._measurement_consumer,
+            unit,
+            description,
+        )
 
     def create_observable_gauge(
         self, name, callback, unit=None, description=None
     ) -> APIObservableGauge:
-        return ObservableGauge(
-            self._instrumentation_info, name, callback, unit, description
+
+        instrument = ObservableGauge(
+            self._instrumentation_info,
+            name,
+            self._measurement_consumer,
+            callback,
+            unit,
+            description,
         )
 
+        self._measurement_consumer.register_asynchronous_instrument(instrument)
+
+        return instrument
```

Comment:
Does it make sense to remove the return statement for asynchronous instruments?

Reply:
We could but we'd need to update the API package as well. I'm curious if other languages return an instrument or not.

Reply:
Java returns NOOP

Reply:
Can someone make a separate issue for this documenting what other languages are doing? I had a look at Java and the builder will return an instrument handle: https://github.com/open-telemetry/opentelemetry-java/blob/5ff5f3f494fe6958574acff0e8c59bc013b69dad/api/all/src/main/java/io/opentelemetry/api/metrics/LongCounterBuilder.java#L54